substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/pecoff/cv/CVSymbolSubrecord.java

```diff
@@ -0,0 +1,648 @@
+            this.offset = offset;
+            this.segment = segment;
+            this.flags = flags;
+        }
+
+        CVSymbolGProc32Record(CVDebugInfo cvDebugInfo, String name, int pparent, int pend, int pnext, int proclen, int debugStart, int debugEnd, int typeIndex, int offset, short segment, byte flags) {
+            this(cvDebugInfo, CVDebugConstants.S_GPROC32, name, pparent, pend, pnext, proclen, debugStart, debugEnd, typeIndex, offset, segment, flags);
+        }
+
+        @Override
+        protected int computeSize(int initialPos) {
+            return computeContents(null, initialPos);
+        }
+
+        @Override
+        protected int computeContents(byte[] buffer, int initialPos) {
+            int pos = CVUtil.putInt(pparent, buffer, initialPos);
+            pos = CVUtil.putInt(pend, buffer, pos);
+            pos = CVUtil.putInt(pnext, buffer, pos);
+            pos = CVUtil.putInt(proclen, buffer, pos);
+            pos = CVUtil.putInt(debugStart, buffer, pos);
+            pos = CVUtil.putInt(debugEnd, buffer, pos);
+            pos = CVUtil.putInt(typeIndex, buffer, pos);
+            if (buffer == null) {
+                cvDebugInfo.getCVSymbolSection().getOwner().createDefinedSymbol(name, getTextSection(), offset, proclen, true, true);
+            }
+            if (buffer != null) {
+                // CVUtil.debug("CVSymbolGProc32Record() adding SECREL reloc at pos=0x%x for func=%s
+                // addr=0x%x\n", pos, name, offset);
+                cvDebugInfo.getCVSymbolSection().markRelocationSite(pos, 4, ObjectFile.RelocationKind.SECREL, name, false, 1L);
+            }
+            pos = CVUtil.putInt(0, buffer, pos);
+            if (buffer != null) {
+                // CVUtil.debug("CVSymbolGProc32Record() adding SECTION reloc at pos=0x%x for
+                // func=%s addr=0x%x\n", pos, name, offset);
+                cvDebugInfo.getCVSymbolSection().markRelocationSite(pos, 2, ObjectFile.RelocationKind.SECTION, name, false, 1L);
+            }
+            pos = CVUtil.putShort((short) 0, buffer, pos);
+            pos = CVUtil.putByte(flags, buffer, pos);
+            pos = CVUtil.putUTF8StringBytes(name, buffer, pos);
+            return pos;
+        }
+
+        private ObjectFile.Element getTextSection() {
+            if (textSection == null) {
+                textSection = cvDebugInfo.getCVSymbolSection().getOwner().elementForName(SectionName.TEXT.getFormatDependentName(ObjectFile.Format.PECOFF));
+            }
+            return textSection;
+        }
+
+        @Override
+        public String toString() {
+            cvDebugInfo.getCVSymbolSection().debugAddress = offset;
+            return String.format("S_GPROC32   name=%s parent=%d debugstart=0x%x debugend=0x%x len=0x%x offset=0x%x type=0x%x flags=0x%x)", name, pparent, debugStart, debugEnd, proclen, offset,
+                            typeIndex,
+                            flags);
+        }
+    }
+
+    /*-
```

Comment:
Is there really a need for this? If not delete it. If so the maybe mark it as unused because the commented out code won't pass the style formatter without getting munged into a syntactic mess.

Reply:
These commented out sections implement records that will probably be used once type information is added.  I can mark them unused if that's preferable.

TIP: Prefixing the command block with "/*-" (as seen here) avoids the munging.  
https://www.oracle.com/java/technologies/javase/codeconventions-comments.html#680

Reply:
Oh, that tip is good to know. Thanks!
I think it would be better to leave this marked unused with a comment to explain why it will eventually be needed.

Reply:
I've added a comment.